gokart/task.py

```diff
@@ -359,7 +359,12 @@ def dump(self, obj: Any, target: Union[None, str, TargetOnKart] = None) -> None:
         PandasTypeConfigMap().check(obj, task_namespace=self.task_namespace)
         if self.fail_on_empty_dump and isinstance(obj, pd.DataFrame):
             assert not obj.empty
-        self._get_output_target(target).dump(obj, lock_at_dump=self._lock_at_dump)
+
+        params = []
+        for param_name, param_obj in self.get_params():
+            params.append((param_name, self.param_kwargs[param_name], param_obj)) if param_obj.significant else None
```

Comment:
immutability is the matter for readability.
```suggestion
        params = [(param_name, self.param_kwargs[param_name], param_obj) for param_name, param_obj in self.get_params() if param_obj.significant else None]
```